R/pp_check.R

```diff
@@ -205,20 +206,34 @@ print.performance_pp_check <- function(x, verbose = TRUE, ...) {
   original <- x$y
   replicated <- x[which(names(x) != "y")]
 
-  if (min(replicated) > min(original)) {
-    if (verbose) {
-      insight::print_color(
-        insight::format_message("Warning: Minimum value of original data is not included in the replicated data.", "Model may not capture the variation of the data."),
-        "red"
-      )
-    }
-  }
+  if (isTRUE(verbose)) {
+    if (is.numeric(original)) {
+      if (min(replicated) > min(original)) {
+        insight::print_color(
+          insight::format_message("Warning: Minimum value of original data is not included in the replicated data.", "Model may not capture the variation of the data."),
+          "red"
+        )
+      }
+
+      if (max(replicated) < max(original)) {
+        insight::print_color(
+          insight::format_message("Warning: Maximum value of original data is not included in the replicated data.", "Model may not capture the variation of the data."),
+          "red"
+        )
+      }
+    } else {
+      sapply(
+        unique(original),
+        function(i) {
+          if (! i %in% replicated) {
```

Comment:
Couldn't we just collect all the levels that are not in "replicated", and then yield just one warning, like "Level(s) a, b, c are not included..."

Reply:
good point!